src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,480 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.Telemetry;
+using Microsoft.CodeAnalysis.MSBuild;
+using File = System.IO.File;
+using Path = System.IO.Path;
+
+namespace AnalyzerRunner
+{
+    /// <summary>
+    /// AnalyzerRunner is a tool that will analyze a solution, find diagnostics in it and will print out the number of
+    /// diagnostics it could find. This is useful to easily test performance without having the overhead of visual
+    /// studio running.
+    /// </summary>
+    class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var options = Options.Create(args);
+            if (options == null)
+            {
+                return;
+            }
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+            Console.CancelKeyPress +=
+                (sender, e) =>
+                {
+                    e.Cancel = true;
+                    cts.Cancel();
+                };
+
+            var analyzerAssemblies = GetAnalyzerAssemblies(options.AnalyzerPath);
+            var analyzers = GetAllAnalyzers(analyzerAssemblies);
+            analyzers = FilterAnalyzers(analyzers, options).ToImmutableArray();
+
+            if (analyzers.Length == 0)
+            {
+                Utilities.WriteLine("No analyzers found", ConsoleColor.Red);
+                Utilities.PrintHelp();
+                return;
+            }
+            var cancellationToken = cts.Token;
+
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            using (MSBuildWorkspace workspace = MSBuildWorkspace.Create())
+            {
+                Solution solution = await workspace.OpenSolutionAsync(options.SolutionPath, cancellationToken).ConfigureAwait(false);
```

Comment:
OK. Where can/should we handle this? Do you mean that OpenSolutionAsync provides diagnostics we would like to display?

I thought that errors will be handled on the compilation step, and this could be enough.

---
In reply to: [150124643](https://github.com/dotnet/roslyn/pull/23087#discussion_r150124643) [](ancestors = 150124643)